dallinger/experiment_server/experiment_server.py

```diff
@@ -86,6 +86,21 @@ def Experiment(args):
 else:
     app.register_blueprint(extra_routes)
 
+# Register extra_routes defined on experiment class
+try:
+    exp_klass = experiment.load()
+    bp = exp_klass.experiment_routes
+    routes = experiment.EXPERIMENT_ROUTE_REGISTRATIONS
+    for rule in routes:
+        route = routes[rule]
+        route_func = getattr(exp_klass, route["func_name"], None)
+        if route_func is not None:
+            bp.route(rule, *route["args"], **route["kwargs"])(route_func)
+    if routes:
+        app.register_blueprint(bp)
+except ImportError:
+    pass
+
 # Ideally, we'd only load recruiter routes if the recruiter is active, but
```

Comment:
Might your decorator-based deferral solve the problem I encountered with recruiter-specific routes as well? I guess we'd need to switch to a model where we put recruiter classes in separate modules so we could import only the one we're going to use at run-time. Just want to keep this idea bubbling in the background while we're working on recruiter infrastructure.

Reply:
It seems like you've got a workaround for these recruiter route issues, but if not a similar deferral like I did for experiment routes and dashboard routes could be useful there.